### PR TITLE
Refactor RamValue level computations into a RamAnalysis

### PR DESCRIPTION
**Pull request by brodyf**
- Related to #541
- Follow on from #858

This is part of tidying up the RAM representation => level information for RamValues (which is primarily used to determine the earliest point in a loop nest that the value can be evaluated) is now computed with a RamAnalysis class.

### CODE REVIEW

File: src/RamConditionLevel.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#include "RamValueLevel.h"
+#include "RamVisitor.h"
+
+namespace souffle {
+
+/** Get level of condition (which for-loop of a query) */
+size_t RamConditionLevelAnalysis::getLevel(const RamCondition* condition) const {
+    // visitor
+    class ConditionLevelVisitor : public RamVisitor<size_t> {
+        RamValueLevelAnalysis* rvla;
```
Comment (mmcgr) [inline src/RamConditionLevel.cpp]: Should this be created somewhere?

File: src/RamOperation.h [hunk 2]
```diff
@@ -397,6 +400,10 @@ class RamLookup : public RamSearch {
  * Aggregation
  */
 class RamAggregate : public RamSearch {
+    RamConstValueAnalysis* rcva;
```
Comment (mmcgr) [inline src/RamOperation.h]: Should these be created somewhere?

File: src/RamConditionLevel.cpp (hunk 1 above)
Comment (brodyf) [reply src/RamConditionLevel.cpp]: Fixed in #866

File: src/RamOperation.h (hunk 2 above)
Comment (brodyf) [reply src/RamOperation.h]: Fixed in #866, had to duplicate some code until aggregates get refactored

**Pull request merged**